makerdao/governance-portal-v2
# Fix/dark mode flash

### Link to Clubhouse story

https://app.clubhouse.io/dux-makerdao/story/364/page-flashes-on-reload-if-in-dark-mode

### What does this PR do?

Adds a next.js custom document class so we can fetch the last-used color mode from localStorage before first paint. This fixes the quick flash of lightmode before switching to darkmode. Also adds fixes for the same issue relating to the skeleton component and the hero image on the landing page.

### Steps for testing:

1. Switch to dark mode
2. cmd + r or refresh the page
3. you shouldn't see the light mode flash

### Screenshots (if relevant):
![Screen Recording 2021-07-30 at 3 47 51 PM](https://user-images.githubusercontent.com/13105602/127714995-68085a55-cd2c-4abd-aeb1-24fa987c8087.gif)

### Additional information:

adds the sha hash of the color mode script to the content security policy to allow the script to run.


### Random GIF:
![ysat](https://media.giphy.com/media/cdlr2QaQ4o4lEtiXkW/giphy.gif)
